utils/config/user.go

```diff
@@ -0,0 +1,62 @@
+package config
+
+import (
+	"fmt"
+
+	"golang.org/x/oauth2"
+)
+
+// GetUser will return a user from the config by name
+func (c *Config) GetUser(user string) (*User, error) {
+	if i, ok := c.checkUserExists(user); ok {
+		return &c.Users[i], nil
+	}
+	return nil, fmt.Errorf("no user found")
+}
+
+// NewUser will add a new user to the config
+func (c *Config) NewUser(user UserConfig) error {
+	if user.Name == "" {
+		return fmt.Errorf("user name not provided")
+	}
+	if user.Grant == nil {
+		user.Grant = &oauth2.Token{}
+	}
+	if _, ok := c.checkUserExists(user.Name); !ok {
+		c.Users = append(c.Users, User{
+			Name:       user.Name,
+			UserConfig: user,
+		})
+		return nil
+	} else {
+		return fmt.Errorf("user %s already exists", user.Name)
+	}
+}
+
+// UpdateUser updates a user within the config
+func (c *Config) UpdateUser(user UserConfig) error {
+	if i, ok := c.checkUserExists(user.Name); ok {
```

Comment:
same as with context - might be useful to check they haven't smashed the user's name here? Just check it's not empty (although this isn't as important)